PWGCF/TwoParticleCorrelations/Tasks/lambdaR2Correlation.cxx

```diff
@@ -336,6 +385,66 @@ struct lambdaCorrTableProducer {
     }
   }
 
+  template <ParticleType v0part, PidType pos_prong, PidType neg_prong, MassWindowType masswin, typename C, typename V, typename T>
+  void selV0Particle(C const& collision, V const& v0track, T const& tracks)
+  {
+
+    // initialize variables
+    auto postrack = v0track.template posTrack_as<T>();
+    auto negtrack = v0track.template negTrack_as<T>();
+    float mass = 0., rap = 0.;
+
+    if (v0part == kLambda) {
+      mass = v0track.mLambda();
+    }
+    if (v0part == kAntiLambda) {
+      mass = v0track.mAntiLambda();
+    }
+
+    if (cfg_do_eta_analysis) {
+      rap = v0track.eta();
+    } else {
+      rap = v0track.yLambda();
+    }
+
+    // apply mass window selection [global]
+    if ((fabs(mass - MassLambda0) >= cfg_lambda_mass_window) || (fabs(v0track.mK0Short() - MassK0Short) <= cfg_kshort_rej)) {
+      return;
+    }
+
+    // apply daughter particle id
+    if (!selPIDTrack<pos_prong>(postrack) || !selPIDTrack<neg_prong>(negtrack)) {
+      return;
+    }
+
+    // apply kinematic acceptance
+    if (v0track.pt() <= cfg_v0_pt_min || v0track.pt() >= cfg_v0_pt_max) {
+      return;
+    }
+
+    if (cfg_do_eta_analysis && (fabs(v0track.eta()) >= cfg_v0_rap_max)) {
+      return;
+    } else if (fabs(v0track.yLambda()) >= cfg_v0_rap_max) {
+      return;
+    }
+
+    if (v0part == kLambda) {
+      histos.fill(HIST("QA_Checks/h1d_lambda_mass"), v0track.mLambda());
+    }
+
+    if (v0part == kAntiLambda) {
+      histos.fill(HIST("QA_Checks/h1d_lambda_mass"), v0track.mAntiLambda());
+    }
+
+    // apply further mass window selection [central, left, right]
+    if (mass > mass_map_min[masswin] && mass < mass_map_max[masswin]) {
+      if (masswin == kCentralWindow) {
+        fillQALambda<v0part>(collision, v0track, tracks);
+      }
+      lambdaTrackTable(lambdaCollisionTable.lastIndex(), v0track.pt(), rap, v0track.phi(), mass, postrack.index(), negtrack.index(), v0part, masswin);
+    }
```

Comment:
A loop on the three mass windows with a break when found will allow to get the proper mass window